src/lib/lib_cron.py

```diff
@@ -63,11 +63,18 @@ def install_cronjob(self):
 
             logrotate_unit = hookenv.local_unit()
             python_venv_path = os.getcwd().replace("charm", "") + ".venv/bin/python3"
+            if hookenv.has_juju_version("3.0"):
+                juju_exec = "juju-exec"
```

Comment:
The behavior of `juju exec` is different, it will not wait 5m for task to complete as `juju run` do. 
```bash
$ juju exec --help | grep wait
--wait  (= 0s)
    Maximum wait time for a task to complete 
$ juju2 run --help | grep timeout
--timeout  (= 5m0s) 
```
Is that ok? Or should we use `--wait 5m`?

Reply:
True, the task should run in background, but we are checking the output or not? Maybe you are right and it's not needed. Feel free to resolve this if you think it's ok.